### PR TITLE
L6 SAW changes

### PR DESCRIPTION
**Pull request by MMMiracles**
:cl: MMMiracles
tweak: The SAW has been revamped slightly. It now requires a free hand to fire, but includes 4 new ammo variants along with a TC cost decrease for both the gun and the ammo boxes.
/:cl:

Its essentially a giant automatic revolver with a huge ammo supply how are you firing this ONE-HANDED YOU'RE GOING TO GET ARTHRITIS IN YOUR WRIST.

I'll look into adding some sort of attachments for guns that can be applied to toggle nodrop via action button, like some sort of special exo-grip that attaches itself to the user's wrist to prevent dropping.

### CODE REVIEW

**Comment by tkdrg** [comment]
I guess you could saw it's a...
two-handed saw

**Comment by Cebasiz** [comment]
probably a good change

to be honest i wish the non-bulldog weapons were better. nobody ever uses them.

C-20 should be like 7, the rifle should be like 10-12.

SAW is seems like a liability because ops don't really have problems shooting people to death. it's the attrition of being shot at that's the problem. an eshield plus the free bulldog is better in every case than this thing, even with the price nerf.

**Comment by Zilenan91** [comment]
Yeah all the weapons need to be cheaper. They're way too expensive for what they do when Ops already have automatic revolvers.

**Comment by PKPenguin321** [comment]
i like it
remind me, does this come with a syndicate pin installed?

**Comment by swankcookie** [comment]
yep

**Comment by Fox-McCloud** [comment]
Is this item even purchased much, at all? If not, this is going to kill it off because you can't go shield+this.

**Comment by Cebasiz** [comment]
nah fox, i've seen one purchased in years of playtime.

there was a brief renaissance of use while nodrops were a thing, as you could equip your automatic rifleman with a nodrop and be sure it wouldn't be stolen by someone who slipped him in the halls or something, but after the shitstorm of criticism about nodrop implants they were consigned back to the dustbin.

**Comment by Steelpoint** [comment]
Historically the only time you see the SAW used was during admin spawned Nuke Op rounds and they just spawn a few down for no reason.

It is just too massive a risk to buy a SAW, considering its raw power if you were to lose it to a disarm the high damage output is enough to mitigate the Op's armour and kill them.

Not to mention the high cost forced you to skip out on a lot of weapons and other items that honestly made the gun unattractive to buy.

Now days what's the point of buying a 60 brute machinegun for half your TC's when you can just buy a M90 that'll crit almost anyone in a single burst fire.

**Comment by Fox-McCloud** [comment]
Alright---in that case, it doesn't make much sense to nerf this to hell then, for a mere 5 TC drop in price.

**Comment by Cebasiz** [comment]
just lower the cost of other syndi weapons. how hard is it to do anyway? shouldn't it be editing basically a few lines of text somewhere?

**Comment by Steelpoint** [comment]
The cost of items is in the traitor uplink area of the code, its very easy to do.

**Comment by KorPhaeron** [comment]
>high damage output is enough to mitigate the Op's armour and kill them.

It's syndi pinned.

**Comment by Cebasiz** [comment]
that's true kor, but just having it picked up by greyshirt #108347209583720857028 screws the team in a way that losing a bulldog doesn't.

one of the chief advantages of investing in a syndiborg, mech, or utility items instead is that they aren't likely to get looted after you get stunned for two seconds by a stray taser shot or a tile of water.

**Comment by KorPhaeron** [comment]
Yeah I was gonna say "make it cheaper" but then I remembered the gun literally instant kills you straight past crit so nevermind

**Comment by Cebasiz** [comment]
is that even a thing?
 
wouldn't you falling over to go into crit prevent the last 3~ shots from hitting you?

as far as i'm concerned the thing has marginal utility in that it has an almost bottomless ammo drum, so you can keep firing through chokepoints to stop people from entering (or face instantcrit) but the cost, and other factors keep it from ever being used.

the ammo is also murderously expensive if i remember right, and even the box it has doesn't last forever with it's 5 shot burst(?)

**Comment by KorPhaeron** [comment]
The shots hit fast enough that they kill you before you hit the ground

**Comment by Cebasiz** [comment]
ah, oh well. i don't play nuke enough to care i guess, but this change arguably makes it worse and it seems like the wrong direction to in to me.

eh

**Comment by KorPhaeron** [comment]
It was bad and never bought before and it'll still be bad and never bought now I wouldnt worry about it too much.

Just a gun that fires generic projectiles, not like we're missing out on any grand potential

**Comment by Cebasiz** [comment]
yeah, but it could be good. people say that nukeop meta is too stale, fixing up the non-bulldog weapons would change it up a little bit.

besides, it seems like the change(s) that could make these weapons good are semi-trivial to make.

this PR probably required more effort than the balance changes we're talking about.

**Comment by KorPhaeron** [comment]
It might be interesting if it could supress people but that'd be way more effort yeah

**Comment by Dorsisdwarf** [comment]
Bulldogs being insane is my fault

I did a basic damage conversion without considering the spread being a weakness of buckshot.

They should start with taser slugs again

**Comment by Cebasiz** [comment]
starting with taser slugs is hardly a nerf, considering bulldog slug mags are like what 2tc?

bulldogs aren't the problem, the cost of the competition is the problem.

bulldogs are used not because they are the best but because they are competitive and cheaper than the (only slight superior) upgrade by huge margins.

**Comment by MMMiracles** [comment]
I could look into giving each operative his own personal voucher to choose between a few iconic weapons like the c20 or bulldog, along with nerfing weapon prices in general for more variety to break away from static meta strats.

Thats a whole other thing for a PR though.

**Comment by CPTANT** [comment]
The weapon is still very boring like this. Since it still feels like a revolver with more ammo.

I would suggest remaking it to be full auto only and then rebalance it around that.

that way the weapon would be actually thematically different.

**Comment by MMMiracles** [comment]
You can't really make a weapon full-auto. Everything we have is essentially semi-auto so it becomes how fast you can click than anything.

**Comment by CPTANT** [comment]
We have plenty of weapons that fire in 3 shot burts.

**Comment by MMMiracles** [comment]
The saw can do 5-burst or semi-auto already.

**Comment by CPTANT** [comment]
Yes and because the burst fire mode sucks so everyone uses the semi auto only. In semi auto it is just a revolver with more ammo.

Which is why removing the semi auto mode and rebalancing the gun around that would really help to make it a more distinct and interesting weapon.

**Comment by MMMiracles** [comment]
I'll look into that later, I guess.

**Comment by Nanotransistor** [comment]
I dont see anything bad with this. Yeah you cant use a shield, but with the extra TC you can buy adrenals.

**Comment by MMMiracles** [comment]
Ok I tried starting an exo-grip. It compiles but the action button won't come up and my requests for help at coderbus were not fruitful.

send help

**Comment by Steelpoint** [comment]
This weapon is useless either way.

Its a TC trap for dumb Nuke Ops and the game cannot simulate effective support weapons like what the SAW should be.

I would suggest it simply be removed from the Ops buy menu.

**Comment by Dorsisdwarf** [comment]
>starting with taser slugs is hardly a nerf, considering bulldog slug mags are like what 2tc?

>bulldogs aren't the problem, the cost of the competition is the problem.

>bulldogs are used not because they are the best but because they are competitive and cheaper than the (only slightly superior) upgrade by huge margins.

Bulldogs are used because I miscalcuated and made them fucking free 8-shot revolvers instead of giving them their taser slugs back.

**Comment by tkdrg** [comment]
whitespace conflicts, see orange's post

**Comment by optimumtact** [comment]
update to latest master
git rebase -Xignore-all-whitespace upstream/master

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -148,14 +148,14 @@ var/list/uplink_items = list()
 	surplus = 50
 
 /datum/uplink_item/dangerous/carbine/unrestricted
-	item = /obj/item/weapon/gun/projectile/automatic/m90/unrestricted
+	item = /obj/itemdweapon/gun/projectile/automatic/m90/unrestricted
```
Comment (Thunder12345) [inline code/datums/uplink_item.dm]: This is why it won't compile

Comment (MMMiracles) [inline code/datums/uplink_item.dm]: that did i even manage that but ok

**Comment by MMMiracles** [comment]
You can remove the Work In Progress tab, its done.

**Comment by tkdrg** [comment]
I took the freedom of adding the two-handed thing to your changelog;

**Pull request merged**